src/reducers/db.js

```diff
@@ -32,6 +33,24 @@ export const dbReducer = (state = {}, action) => {
         };
       }
       break;
+    case OPTIMISTIC_UPDATE_PHONE:
+      const location = state[action.payload.locationId];
+      const idx = location.Phones.map( phone => phone.id).indexOf(action.payload.phoneId)
```

Comment:
findIndex could save an iteration here.

Reply:
Fixed in cdf652af033e7b9e18f52372d76b55de2ae0efb7